File: src/main/java/spoj/other/Flamaster.cpp
///////////////////////////////////////////////////
/// NOTES: ////////////////////////////////////////
/// TASK ID = 506, NAME: FLAMASTER ////////////////
/// link: https://pl.spoj.com/problems/FLAMASTE ///
/// TIME RESULT FOR THIS SOLUTION = 0,00s /////////
/// SOURCE CODE LENGTH LIMIT = 2000B //////////////
///////////////////////////////////////////////////

#include <iostream>
#include <string>
using namespace std;

string getShortenedString(string input){
string result = "";
int t, length, i, j, k, count;
length = input.length();
for (j = 0; j < length - 2; j++) {
	result.push_back(input[j]);
	if (input[j] == input[j + 1] && input[j + 1] == input[j + 2]) {
	k = j + 1;
	count = 1;
	while (input[j] == input[k]) {
		count++;
		k++;
		if (k == length) break;
	}
	result.append(to_string(count));
	j = j + count - 1;
	}
}
if (j == length - 2){
	result.push_back(input[j]);
	result.push_back(input[j + 1]);
} else if (j == length - 1) result.push_back(input[j]);

return result;
}

int main() {
	int t, length, i, j, k, count;
	string input, result;
	cin >> t;
	for (int i = 0; i < t; i++) {
		cin >> input;
        result = getShortenedString(input);
		cout << result << "\n";
	}
	return 0;
}

File: src/main/java/spoj/Flamaster.java
///////////////////////////////////////////////////
/// NOTES: ////////////////////////////////////////
/// TASK ID = 506, NAME: FLAMASTER ////////////////
/// link: https://pl.spoj.com/problems/FLAMASTE ///
/// TIME RESULT FOR THIS SOLUTION = 0,06s /////////
/// SOURCE CODE LENGTH LIMIT = 2000B //////////////
///////////////////////////////////////////////////
package spoj;

import java.io.*;

public class Flamaster {

    private static String getShortenedString(String inputString) {
        StringBuilder strb = new StringBuilder();
        char[] string = inputString.toCharArray();
        int length = string.length;
        int j;
        for (j = 0; j < length - 2; j++) {
            strb.append(string[j]);
            if (string[j] == string[j + 1] && string[j + 1] == string[j + 2]) {
                int k = j + 1;
                int count = 1;
                while (string[j] == string[k]) {
                    count++;
                    k++;
                    if (k == length) break;
                }
                strb.append(count);
                j = j + count - 1;
            }
        }
        if (j == length - 2) {
            strb.append(string[j]).append(string[j + 1]);
        } else if (j == length - 1) {
            strb.append(string[j]);
        }

        return strb.toString();
    }

    public static void main(String[] arg) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        OutputStream out = new BufferedOutputStream(System.out);
        StringBuilder strb = new StringBuilder();

        int tests = Integer.parseInt(br.readLine());
        for (int i = 0; i < tests; i++) {
            String input = br.readLine();
            strb.append(getShortenedString(input)).append("\n");
        }
        out.write(strb.toString().getBytes());
        out.flush();
    }
}